docs/docs/09-rfcs/2023-03-08-wing-preview-environment.md

```diff
@@ -0,0 +1,364 @@
+| failed.main.w   | ❌ Failed ([logs](https://wing.cloud/dashboard/-account/-project/-branch/logs/build/))     |                                                               |                                                                                                                                                        | Jul 31, 2023 8:01am |
+| building.main.w | 🔄 Building ([logs](https://wing.cloud/dashboard/-account/-project/-branch/logs/build/))   |                                                               |                                                                                                                                                        | Jul 31, 2023 8:01am |
+| stale.main.w    | ⏸️ Stale ([lean more](https://wing.cloud/dashboard/-account/-project/-branch/logs/build/)) |                                                               |                                                                                                                                                        | Jul 31, 2023 8:01am |
+
+#### Preview Environment Environment Variables (Secrets) (#9)
+
+The project management dashboard will include a section to manage the secrets of the project. These secrets will be available during the Wing compilation.
+
+#### Run Tests Automatically Upon PR Creation and Code Changes (#10)
+
+Upon PR creation and code changes, Wing Cloud Preview Environments will automatically run all tests defined for the entry point.
+For each test run, a new simulator will be created and will be destroyed upon test completion.
+The test results will be available in the PR comment with links to the tests logs (`https://wing.cloud/dashboard/-account/-project/-branch/logs/tests/<test-name>`)
+
+#### Running Tests Manually From Wing Console (#11)
+
+Running tests on your environment is easy using the Wing Console tests UI.
+Each test will reload the simulator to create a clean environment.
+
+Once the test is over it will **not** clean up the simulator data, so you can debug the environment and share outputs with your teammates.
+
+#### User-Controlled Environments termination (#12)
+
+Closing a PR will terminate the corresponding preview environments.
+The PR comment will indicate the termination of each preview environment and the URLs won't be valid anymore.
+
+#### Self-Cleaning Environments Mechanism (#13)
+
+To ensure efficient resource utilization:
+
+1. Preview environments associated with stale PRs (without access or code changes for over 30 days) will automatically deactivate.
+2. The PR's preview environment comment will indicate its inactive status.
+3. Changes in code will trigger redeployment for all related preview environments in this PR.
+
```

Comment:
missing: how do users access application endpoints, such as websites and api endpoints?